src/ui/public/agg_types/buckets/geo_hash.js

```diff
@@ -45,19 +69,17 @@ export default function GeoHashAggDefinition(Private, config) {
       },
       {
         name: 'precision',
-        default: defaultPrecision,
         editor: precisionTemplate,
         controller: function ($scope) {
-          $scope.$watchMulti([
-            'agg.params.autoPrecision',
-            'outputAgg.params.precision'
-          ], function (cur, prev) {
-            if (cur[1]) $scope.agg.params.precision = cur[1];
-          });
         },
         deserialize: getPrecision,
         write: function (aggConfig, output) {
-          output.params.precision = getPrecision(aggConfig.params.precision);
+          let currZoom = aggConfig.vis.params.mapZoom;
```

Comment:
Not a fan of this function. I have to figure out whether the `aggConfig`, `vis`, or `uiState` has the proper zoom, to base the precision off of. This happens differently whether you load up the dashboard with the same vis, refresh on the vis page, or set it by zooming or panning on the map